src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala

```diff
@@ -88,10 +99,24 @@ class TimeSeriesRelation(config: RelationConfig)(val sqlContext: SQLContext)
   override def getStreams(filters: Array[Filter])(
       client: GenericClient[IO, Nothing],
       limit: Option[Long],
-      numPartitions: StatusCode): Seq[fs2.Stream[IO, TimeSeries]] =
-    Seq(
-      config.limit.map(client.timeSeries.listWithLimit(_)).getOrElse(client.timeSeries.list)
-    )
+      numPartitions: StatusCode): Seq[fs2.Stream[IO, TimeSeries]] = {
+
+    val fieldNames = Array("assetId")
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val shouldGetAllRows = shouldGetAll(pushdownFilterExpression, fieldNames)
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+    val assetIds = filtersAsMaps.flatMap(m => m.get("assetId").map(_.toLong))
+    if (assetIds.isEmpty || shouldGetAllRows) {
```

Comment:
Should use `limit` not `config.limit`, and `client.timeSeries.filter` if it's not present:
```scala
    if (filtersAsMaps.isEmpty || shouldGetAllRows) {
      Seq(
        limit
          .map(client.timeSeries.listWithLimit)
          .getOrElse(client.timeSeries.list)
      )
    } else {
      val filter = TimeSeriesFilter(assetIds = assetIds)
      Seq(
        limit
          .map(l => client.timeSeries.filterWithLimit(filter, l))
          .getOrElse(client.timeSeries.filter(filter))
      )
    }
```